core/src/toga/screen.py

```diff
@@ -0,0 +1,13 @@
+class Screen:
+    def __new__(cls, _impl):
+        if _impl.native in _impl._instances and _impl.interface is not None:
+            return _impl._instances[_impl.native].interface
```

Comment:
This seems unnecessary (and a violation of the spirit of the interface/impl layer boundary). If the only mechanism for creating an interface-level instance is as a result of the backend instantiating an instance, that instance can call `toga.Screen(self)` in its constructor, rather than setting interface to None and then backfilling.